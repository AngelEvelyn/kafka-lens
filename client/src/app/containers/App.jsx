import React from 'react';

import Main from './Main.jsx';

// const App = props => {
//   return <Main />;
// };
class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};

    // bind methods here
  }

  render() {
    return <Main />;
  }
}

export default App;
