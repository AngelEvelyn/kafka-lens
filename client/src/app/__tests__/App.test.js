import React from 'react';
import { shallow } from 'enzyme';

import App from '../containers/App.jsx';

describe('App.js unit tests', () => {
  let wrapper;

  // * MOUNT MAIN COMPONENT BEFORE ALL TESTS
  beforeAll(() => {
    wrapper = shallow(<App />);
  });

  it('should render without errors', () => {
    expect(wrapper.exists('.main-div')).toEqual(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a single div with class: main-div', () => {
    expect(wrapper.exists('.main-div')).toEqual(true);
  });

  it('should render <ConnectionPage/>', () => {
    expect(wrapper.exists('ConnectionPage')).toEqual(true);
  });

  it('should render <BrowserRouter/> if this.state.connected is true', () => {
    const instance = wrapper.instance();
    expect(instance.state.connected).toBe(null);
    instance.setState({ connected: true });
    expect(instance.state.connected).toBe(true);
    expect(wrapper.exists('BrowserRouter')).toEqual(true);
  });
});
