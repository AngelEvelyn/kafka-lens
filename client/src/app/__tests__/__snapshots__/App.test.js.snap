// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js unit tests renders without errors 1`] = `<App />`;
